src/object/tcb.c

```diff
@@ -1363,12 +1407,12 @@ exception_t decodeSetSchedParams(cap_t cap, word_t length, word_t *buffer)
         return EXCEPTION_SYSCALL_ERROR;
     }
 
-    if (!validFaultHandler(fhCap)) {
-        userError("TCB Configure: fault endpoint cap invalid.");
-        current_syscall_error.type = seL4_InvalidCapability;
-        current_syscall_error.invalidCapNumber = 3;
-        return EXCEPTION_SYSCALL_ERROR;
+    /* fault handler */
+    deriveCap_ret_t dc_ret = updateAndCheckHandlerEP(fhCap, length, fhSlot, buffer, 2, 3);
```

Comment:
Add a `#define DECODE_SET_SHED_PARAMS 2` and use this when checking the length above and pass it here instead of a plain number. Also make the comment state if this is an optional parameter or not.